cfme/tests/intelligence/test_reports_with_timelines.py

```diff
@@ -1,16 +1,251 @@
+from cfme.infrastructure.provider import InfraProvider
+from cfme.infrastructure.provider.scvmm import SCVMMProvider
+from cfme.intelligence.reports.reports import CustomReport
+from cfme.markers.env_markers.provider import providers
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.blockers import BZ
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
+from cfme.utils.path import data_path
+from cfme.utils.providers import ProviderFilter
+from cfme.utils.wait import wait_for
 
+all_prov = ProviderFilter(classes=[InfraProvider, CloudProvider])
+excluded = ProviderFilter(classes=[GCEProvider, SCVMMProvider, OpenStackProvider],
+                          inverted=True)
+pytestmark = [
+    pytest.mark.tier(1),
+    pytest.mark.provider(gen_func=providers, filters=[excluded, all_prov]),
+    pytest.mark.usefixtures('setup_provider')
+]
 
-@pytest.mark.manual
-@pytest.mark.parametrize('report', timelines_reports)
-def test_default_reports_with_timelines(report):
-    pass
+TIMELINES_REPORTS = ('hosts', 'vm_operation', 'policy_events', 'policy_events2')
+REPORTS_CRUD_DIR = data_path.join('timeline_reports')
+REPORTS_FILE_NAME = 'timeline_reports.yaml'
+REPORTS_FILE_PATH = REPORTS_CRUD_DIR.join(REPORTS_FILE_NAME)
 
 
-@pytest.mark.manual
-@pytest.mark.parametrize('report', timelines_reports)
-def test_custom_reports_with_timelines(report):
-    pass
+@pytest.fixture
+def new_vm(provider):
+    vm = VM.factory(random_vm_name('timelines', max_length=16), provider)
+    vm.create_on_provider(find_in_cfme=True)
+    logger.info('Fixture new_vm set up! Name: %r Provider: %r', vm.name, vm.provider.name)
+    yield vm
+    vm.cleanup_on_provider()
+
+
+@pytest.fixture
+def control_policy(appliance, new_vm):
+    action = appliance.collections.actions.create(fauxfactory.gen_alpha(),
+                                                  "Tag", dict(tag=("My Company Tags",
+                                                                   "Environment",
+                                                                   "Development")))
+    policy = appliance.collections.policies.create(VMControlPolicy, fauxfactory.gen_alpha())
+    policy.assign_events("VM Power Off")
+    policy.assign_actions_to_event("VM Power Off", action)
+    profile = appliance.collections.policy_profiles.create(fauxfactory.gen_alpha(),
+                                                           policies=[policy])
+
+    yield new_vm.assign_policy_profiles(profile.description)
+    if profile.exists:
+        profile.delete()
+    if policy.exists:
+        policy.delete()
+    if action.exists:
+        action.delete()
+
+
+@pytest.fixture
+def report_policy_event_week():
+    if REPORTS_FILE_PATH.exists():
+        yaml_dict = yaml.load(open(str(REPORTS_FILE_PATH), 'r'))['policy_event_week']
+        report = CustomReport(**yaml_dict)
+        report.create()
+        yield report
+        report.delete()
```

Comment:
**REQUIRED**
```python
if report.exists:
    report.delete()
```

Reply:
done. Should I also check for existance before creating?

Reply:
No, it's no needed.